scripts/incremental/efficiency.py

```diff
@@ -0,0 +1,232 @@
+        data["Commit"].append(commit_prop["hash"][:7])
+        if commit_prop["failed"] == True:
+            data[utils.header_runtime_parent].append(0)
+            data[utils.header_runtime_incr_child].append(0)
+            data[utils.header_runtime_incr_posts_child].append(0)
+            data[utils.header_runtime_incr_posts_rel_child].append(0)
+            data["Changed/Added/Removed functions"].append(0)
+            data["Change in number of race warnings"].append(0)
+            continue
+        parent_info = utils.extract_from_analyzer_log(parentlog)
+        child_info = utils.extract_from_analyzer_log(childlog)
+        child_posts_info = utils.extract_from_analyzer_log(childpostslog)
+        child_posts_rel_info = utils.extract_from_analyzer_log(childpostsrellog)
+        data["Changed/Added/Removed functions"].append(int(child_info["changed"]) + int(child_info["added"]) + int(child_info["removed"]))
+        data[utils.header_runtime_parent].append(float(parent_info["runtime"]))
+        data[utils.header_runtime_incr_child].append(float(child_info["runtime"]))
+        data[utils.header_runtime_incr_posts_child].append(float(child_posts_info["runtime"]))
+        data[utils.header_runtime_incr_posts_rel_child].append(float(child_posts_rel_info["runtime"]))
+        data["Change in number of race warnings"].append(int(child_info["race_warnings"] - int(parent_info["race_warnings"])))
+    return data
+
+def runperprocess(core, from_c, to_c):
+    if not only_collect_results:
+        psutil.Process().cpu_affinity([core])
+    cwd  = os.getcwd()
+    outdir = os.path.join(cwd, 'out')
+    if not only_collect_results:
+        if os.path.exists(outdir) and os.path.isdir(outdir):
+          shutil.rmtree(outdir)
+        analyze_small_commits_in_repo(cwd, outdir, from_c, to_c)
+    data_set = collect_data(outdir)
+    df = pd.DataFrame(data_set)
+    #df.sort_index(inplace=True, key=lambda idx: idx.map(lambda x: int(x.split(":")[0])))
+    print(df)
+    df.to_csv('results.csv', sep =';')
+
+def analyze_chunks_of_commits_in_parallel():
+    avail_phys_cores = psutil.cpu_count(logical=False)
+    allowedcores = avail_phys_cores - 2
+    if not only_collect_results and numcores > allowedcores:
+        print("Not enough physical cores on this maching (exist: ", avail_phys_cores, " allowed: ", allowedcores, ")")
+        exit()
+    # For equal load distribution, choose a processes to core mapping,
+    # use only physical cores and have an equal number of processes per cache.
+    # The layout of physical/logical cores and sharing of caches is machine dependent. To find out use: 'lscpu --all --extended'.
+    # For our test server:
+    coremapping1 = [i for i in range(numcores - numcores//2)]
+    coremapping2 = [i for i in range(avail_phys_cores//2, avail_phys_cores//2 + numcores//2)]
+    coremapping = [coremapping1[i//2] if i%2==0 else coremapping2[i//2] for i in range(len(coremapping1) + len(coremapping2))]
```

Comment:
Since this sounds very specific to the TUM test server and is deep inside the script, it might be good to point out at the beginning of the script in a comment that `analyze_chunks_of_commits_in_parallel` might require this to be adjusted.